### PR TITLE
Edit Colony Details Modal

### PR DESCRIPTION
**Pull request by chinins**
## Description

Edit Colony Details Modal

**New stuff** ✨

- add `EditColonyDetails` dialog, form, permissions, css
- add colony edit details actions
- added 2nd version of `AvatarUploader` css
- added `disabled` state to `AvatarUploader`
- add refetching in colony avatar actions sagas (not working yet!)

**Changes** 🏗

- updated accepted files in `AvatarUploader`

**Deletions** ⚰️

none

## TODO

- fix refetching of colony when logo is uploaded/removed

Resolves DEV-73

Screenshot 2021-01-14 at 16 18 35
Screenshot 2021-01-14 at 16 18 55

### CODE REVIEW

**Comment by chinins** [comment]
Can't rebase branch on latest changes, because it gives me an error when I push: in resolvers passing `null` 4 times instead of 3. I saw this change in Alicja's branch and I assume it should work. I will keep investigating why it gives an error but for now it doesn't impact the review files.

**Comment by rdig** [comment]
Nope, this is on me.

Wait until after I merge #2347 then it will be all fixed

**Comment by rdig** [comment]
Also, your error is because you didn't run `npm i` and `npm run provision` after rebasing.

Alicja's branch did a couple of submodule refreshes.

Anyway, my PR will fix it, but you'll have to run the same install + provision in order to make the linter happy

**Comment by rdig** [comment]
#2347 is now merged so any type of errors that you encountered before are now fixed.

Please rebase on the latest `master`, install modules and run provision.

**Comment by chinins** [comment]
All rebased, ready for the review

File: src/modules/admin/components/Profile/ColonyAvatarUploader.tsx
Comment (rdig) [inline src/modules/admin/components/Profile/ColonyAvatarUploader.tsx:53]: When uploading an image that is not supported (in this case too large):
Screenshot from 2021-01-19 13-50-00

It triggers some sort of error in `AvatarUploadItem`:
Screenshot from 2021-01-19 14-07-09

I know this is not something you introduced, but please investigate its cause. (To me it looks like a bad import)

File: src/modules/dashboard/components/EditColonyDetailsDialog/EditColonyDetailsDialog.tsx
Comment (rdig) [inline src/modules/dashboard/components/EditColonyDetailsDialog/EditColonyDetailsDialog.tsx:36]: I think the name input will have some sort of limit (at least on the client side)

Would you mind finding this out and putting it in (if any)

For reference the domain name was capped at 20 chars.

File: src/modules/dashboard/components/EditColonyDetailsDialog/EditColonyDetailsDialogForm.tsx
Comment (rdig) [inline src/modules/dashboard/components/EditColonyDetailsDialog/EditColonyDetailsDialogForm.tsx:111]: This heading had a different color for the other ones. Please change it to be more dark (see the one below)

Screenshot from 2021-01-19 13-49-29

File: src/modules/dashboard/sagas/colony.ts
Comment (rdig) [inline src/modules/dashboard/sagas/colony.ts:80]: This should be before  you send the "success" action

**Review by rdig — APPROVED** [review]
Minor things to change, but all in all good work. :100: 

I'm pre-approving it.

File: src/modules/admin/components/Profile/ColonyAvatarUploader.tsx
Comment (chinins) [reply src/modules/admin/components/Profile/ColonyAvatarUploader.tsx:53]: i don't have this error - i just see an error UI - as expected

Screenshot 2021-01-19 at 20 50 15

Comment (chinins) [reply src/modules/admin/components/Profile/ColonyAvatarUploader.tsx:53]: this is both for larger & different format errors... not sure how i can replicate.

File: src/modules/dashboard/sagas/colony.ts
Comment (chinins) [reply src/modules/dashboard/sagas/colony.ts:80]: Moved it to before & it's still not working. Maybe something with my environment... Is it working on your machine?

File: src/modules/core/components/AvatarUploader/AvatarUploader.css
Comment (ArmandoGraterol) [inline src/modules/core/components/AvatarUploader/AvatarUploader.css:16]: You could use `background-color: color-mod(var(----temp-grey-blue-4) alpha(33%));` here

**Review by ArmandoGraterol — DISMISSED** [review]
Solid work :fire: Just some things I noticed:

FireShot Capture 084 - Colony - localhost

I think this should be in a darker, more black, color, at least according to design.

FireShot Capture 087 - Colony - localhost

The background of this section is darker on the design. In the design it's `rgb(232, 236, 245)` I think

FireShot Capture 092 - Colony - localhost

This text should have a `font-weight: 700`, and have a darker color: `rgb(118,116,139)`.

FireShot Capture 095 - Colony - localhost

The text in this input should have a `font-weight: 700`

Overall, just small details, the rest it's pretty good I would say :+1:

File: src/modules/dashboard/sagas/colony.ts
Comment (rdig) [reply src/modules/dashboard/sagas/colony.ts:80]: I just realized this doesn't matter since I'm ripping it out anyway in #2353  :)

So just don't worry about it.

File: src/modules/admin/components/Profile/ColonyAvatarUploader.tsx
Comment (rdig) [reply src/modules/admin/components/Profile/ColonyAvatarUploader.tsx:53]: Bah. Was this on my machine only? I'll have to check again I guess...

**Pull request merged**
